gpytorch/kernels/index_kernel.py

```diff
@@ -82,13 +82,16 @@ def covar_matrix(self):
         return res
 
     def forward(self, i1, i2, **params):
+
+        i1, i2 = torch.LongTensor(i1.long()), torch.LongTensor(i2.long())
```

Comment:
You mentioned explicit cast, but I'm nor sure I understand why you have the `LongTensor` here. This is a nullop on the CPU and breaks on the GPU. 
```suggestion
        i1, i2 = i1.long(), i2.long()
```

Reply:
Oops, you're right. That's just me not knowing pytorch, apparently :).